src/main/java/clinic_management/SetTimePeriod.java

```diff
@@ -23,43 +24,43 @@ public SetTimePeriod(ServiceLocation clinic){
 
     // Setting the number of shifts for a certain time
     public String setEmployees(LocalDate date, int employees) {
-        try{
-            this.clinic.setShift(date, employees);
-            return "Set " + employees + " employees for " + date;
-        } catch(Exception ex) {
-            return null;
-        }
+        this.clinic.setShift(date, employees);
+        return "Set " + employees + " employees for " + date;
     }
 
     // Adding a time period to a clinic if it is not already there
-    public String addTimePeriod(LocalDateTime dateTime){
+    public String addTimePeriod(LocalDateTime dateTime) throws Exception {
         if (this.clinic.shiftAvailable(dateTime.toLocalDate())
                 && !this.clinic.checkTimePeriod(dateTime)){
             int slots = this.clinic.getShiftForDate(dateTime.toLocalDate());
             TimePeriod addedTimePeriod = new TimePeriod(dateTime, slots);
             this.clinic.addTimePeriod(addedTimePeriod, dateTime.toLocalDate());
             return addedTimePeriod.toString();
+        }else if(this.clinic.shiftAvailable(dateTime.toLocalDate())) {
```

Comment:
This should be else if(!this.clinic.shiftAvailable(dateTime.toLocalDate())) ..... NOTICE THE !